test/e2e/snaps/ppom-malicious-erc20-approval.spec.js

```diff
@@ -0,0 +1,592 @@
+      },
+    };
+  });
+
+  await mockServer
+    .forPost()
+    .withJsonBodyIncluding({ method: 'eth_estimateGas' })
+    .thenCallback((req) => {
+      return {
+        statusCode: 200,
+        json: {
+          jsonrpc: '2.0',
+          id: req.body.json.id,
+          result: '0x5cec',
+        },
+      };
+    });
+
+
+
+  await mockServer
+    .forPost()
+    .withJsonBodyIncluding({ method: 'eth_gasPrice' })
+    .thenCallback((req) => {
+      return {
+        statusCode: 200,
+        json: {
+          jsonrpc: '2.0',
+          id: req.body.json.id,
+          result: '0x09184e72a000',
+        },
+      };
+    });
+
+    await mockServer
+    .forGet('https://www.4byte.directory/api/v1/signatures/')
```

Comment:
oh and I think we can remove this one -- was before realizing we were missing a balance call.
This one is specific for contract interactions, but we should not need it for custom screens